filebeat-yml-script.py

```diff
@@ -3,18 +3,27 @@
 from ruamel.yaml import YAML
 import socket
 
+logging.basicConfig(format='%(asctime)s\t%(levelname)s\t%(message)s', level=logging.DEBUG)
+
 # set vars and consts
 
 logzio_url = os.environ["LOGZIO_URL"]
 logzio_url_arr = logzio_url.split(":")
 logzio_token = os.environ["LOGZIO_TOKEN"]
-
 HOST = logzio_url_arr[0]
 PORT = int(logzio_url_arr[1])
-FILEBEAT_CONF_PATH = "/etc/filebeat/filebeat.yml"
+FILEBEAT_CONF_PATH = f"{os.getcwd()}/filebeat.yml"
 SOCKET_TIMEOUT = 3
-
-logging.basicConfig(format='%(asctime)s\t%(levelname)s\t%(message)s', level=logging.DEBUG)
+try:
+    logzio_codec = os.environ['LOGZIO_CODEC'].lower()
+    logzio_codec_list = ["plain", "json"]
+    if logzio_codec not in logzio_codec_list:
+        logging.warning(f"LOGZIO_CODEC={logzio_codec} not supported. Make sure to use one of: {logzio_codec_list}. "
+                        f"Falling back to default LOGZIO_CODEC=plain")
+        logzio_codec = "plain"
+except KeyError:
+    logging.debug("LOGZIO_CODEC is not configured, falling back to default LOGZIO_CODEC=plain")
+    logzio_codec = "plain"
```

Comment:
1) Let's move the try/except up below `logzio_token`
2) before the try let's do `logzio_codec = "plain"` and remove it when we fall back to the default

Reply:
> Let's move the try/except up below logzio_token

Its only a cosmetic change, as this will not affect anything until try/except is between logger definition and main function (if statement in that case). I will change that once the thread is closed.

> before the try let's do logzio_codec = "plain" and remove it when we fall back to the default

Can you explain it please?

Reply:
We can drop try/catch and do `logzio_codec = os.getenv('LOGZIO_CODEC', 'plain')`

Reply:
right, that makes sense.
i have added those changes, please review and approve.
thank you